ksql-engine/src/test/java/io/confluent/ksql/ddl/commands/CreateSourceCommandTest.java

```diff
@@ -146,12 +181,61 @@ public void shouldNotThrowIfTopicDoesExist() {
     when(kafkaTopicClient.isTopicExists(TOPIC_NAME)).thenReturn(true);
 
     // When:
-    new TestCmd("what, no value topic?", statement, ksqlConfig, kafkaTopicClient, serdeOptions);
+    new TestCmd(
+        "what, no value topic?",
+        statement,
+        ksqlConfig,
+        kafkaTopicClient,
+        serdeOptions,
+        serdeFactories
+    );
 
     // Then:
     verify(kafkaTopicClient).isTopicExists(TOPIC_NAME);
   }
 
+  @Test
+  public void shouldThrowIfTopicToCreateDoesExist() {
+    // Given:
+    when(metaStore.getTopic("bob")).thenReturn(topic);
+
+    // Then:
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage("A topic with name 'bob' already exists");
+
+    // When:
+    new TestCmd(
+        "what, no value topic?",
+        statement,
+        ksqlConfig,
+        kafkaTopicClient,
+        serdeOptions,
+        serdeFactories
+    ).createTopic(metaStore);
+  }
+
+  @Test
+  public void shouldNotThrowIfTopicToCreateDoesNotExist() {
```

Comment:
Likewise, can we rename to 

```suggestion
  public void shouldRegisterTopic() {
```

Reply:
Done